mule-extensions-api/src/main/java/org/mule/runtime/extension/api/manifest/ImmutableDescriberManifest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.manifest;
+
+import static java.util.Collections.unmodifiableMap;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import java.util.Map;
+
+/**
+ * Immutable implementation of {@link DescriberManifest}
+ *
+ * @since 1.0
+ */
+final class ImmutableDescriberManifest implements DescriberManifest
```

Comment:
Asked LG, he does not remember having a long debate about this topic.
You say that the class must be here to be able, for example, to deserialize an object. But the extension-api provides many interfaces other than this one, so I guess you will need requiring extension support anyway.
Besides that, what is the reason this class can't be public?
Having a non public class on a api package in counterintuitive. At least it must be in an internal package